src/Sema.zig

```diff
@@ -2672,11 +2672,16 @@ fn zirResolveInferredAlloc(sema: *Sema, block: *Block, inst: Zir.Inst.Index) Com
                 // ZIR handles it later, so we can just use the ty ref here.
                 air_datas[ptr_inst].ty_pl.ty = air_datas[bitcast_inst].ty_op.ty;
 
+                // Unless the block is comptime, `alloc_inferred` always produces
+                // a runtime constant. The final inferred type needs to be
+                // fully resolved so it can be lowered in codegen.
+                try sema.resolveTypeFully(block, ty_src, final_elem_ty);
```

Comment:
@andrewrk Should I just move this above the `if` statement so it is called in all cases? That resolves it and produces no new failed tests, but I wasn't sure if it was safe to do above the other checks that happen in this func.

Reply:
I like how you did it here.

In the future we may need to introduce a new mechanism to queue up all the types that need to be resolved (for codegen purposes) instead of doing them immediately, such that the decl analysis can complete without causing a "dependency loop detected" compile error, and then once decl analysis is complete, it can go through the list of types needed to be resolved.

Anyway point being I think this is fine and we'll likely need to revisit anyway with a larger goal in mind.